.github/workflows/tests.yaml

```diff
@@ -0,0 +1,78 @@
+name: Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Add local bin to PATH
+        run: |
+          echo "${HOME}/.local/bin" >> $GITHUB_PATH
+      - name: Install python package and dependencies
+        run: |
+          sudo apt install -y python3-pip python3-venv
+          python3 -m venv ${HOME}/.venv
+          source ${HOME}/.venv/bin/activate
+          pip install -U pip wheel setuptools
+          pip install -U -r requirements.txt -r requirements-dev.txt
+          pip install -e .
+      - name: Run black
+        run: |
+          source ${HOME}/.venv/bin/activate
+          make test-black
+      - name: Run codespell
+        run: |
+          source ${HOME}/.venv/bin/activate
+          make test-codespell
+      - name: Run flake8
+        run: |
+          source ${HOME}/.venv/bin/activate
+          make test-flake8
+      - name: Run mypy
```

Comment:
Any special reason for not running isort in the GitHub test workflow?

Reply:
Must have missed it, will add :D

Reply:
Good eye btw, fixed!